applications/precompiled_pva/cpp/main.cpp

```diff
@@ -0,0 +1,156 @@
+      auto maybe_input_message = op_input.receive<gxf::Entity>("input");
+      if (!maybe_input_message.has_value()) {
+        HOLOSCAN_LOG_ERROR("Failed to receive input message gxf::Entity");
+        return;
+      }
+      auto input_tensor = maybe_input_message.value().get<holoscan::Tensor>();
+      if (!input_tensor) {
+        HOLOSCAN_LOG_ERROR("Failed to receive holoscan::Tensor from input message gxf::Entity");
+        return;
+      }
+
+      // get handle to underlying nvidia::gxf::Allocator from std::shared_ptr<holoscan::Allocator>
+      auto allocator = nvidia::gxf::Handle<nvidia::gxf::Allocator>::Create(
+          fragment()->executor().context(), allocator_->gxf_cid());
+
+      // cast Holoscan::Tensor to nvidia::gxf::Tensor to use its APIs directly
+      nvidia::gxf::Tensor input_tensor_gxf{input_tensor->dl_ctx()};
+
+      auto out_message = CreateTensorMap(
+          context.context(),
+          allocator.value(),
+          {{"output",
+            nvidia::gxf::MemoryStorageType::kDevice,
+            input_tensor_gxf.shape(),
+            nvidia::gxf::PrimitiveType::kUnsigned8,
+            0,
+            nvidia::gxf::ComputeTrivialStrides(
+                input_tensor_gxf.shape(),
+                nvidia::gxf::PrimitiveTypeSize(nvidia::gxf::PrimitiveType::kUnsigned8))}},
+          false);
+
+      if (!out_message) { std::runtime_error("failed to create out_message"); }
+      const auto output_tensor = out_message.value().get<nvidia::gxf::Tensor>();
+      if (!output_tensor) { std::runtime_error("failed to create out_tensor"); }
+
+      uint8_t* input_tensor_data = static_cast<uint8_t*>(input_tensor->data());
+      uint8_t* output_tensor_data = static_cast<uint8_t*>(output_tensor.value()->pointer());
+      if (output_tensor_data == nullptr) {
+        throw std::runtime_error("Failed to allocate memory for the output image");
+      }
+
+      const int32_t imageWidth{static_cast<int32_t>(input_tensor->shape()[1])};
+      const int32_t imageHeight{static_cast<int32_t>(input_tensor->shape()[0])};
+      const int32_t inputLinePitch{static_cast<int32_t>(input_tensor->shape()[1])};
+      const int32_t outputLinePitch{static_cast<int32_t>(input_tensor->shape()[1])};
+
+      if (!pvaOperatorTask_.isInitialized()) {
+        pvaOperatorTask_.init(imageWidth, imageHeight, inputLinePitch, outputLinePitch);
+      }
+      pvaOperatorTask_.process(input_tensor_data, output_tensor_data);
+      auto result = gxf::Entity(std::move(out_message.value()));
+
+      op_output.emit(result, "output");
+    }
+
+private:
+    Parameter<std::shared_ptr<Allocator>> allocator_;
+    PvaUnsharpMask pvaOperatorTask_;
+};
+} // namespace holoscan::ops
+
+class App : public holoscan::Application
+{
+public:
+    void compose() override
+    {
+        using namespace holoscan;
+
+        uint32_t max_width{1920};
+        uint32_t max_height{1080};
+        int64_t source_block_size = max_width * max_height * 3 * 4;
```

Comment:
How is this value derived?

Reply:
This was due to overestimation of the required buffer size.
Updated it to: "int64_t source_block_size = max_width * max_height * 3;" to allocate memory pool for a single 1920x1080 RGB image.